network/dag/bbolt_tree.go

```diff
@@ -0,0 +1,89 @@
+package dag
+
+import (
+	"encoding/binary"
+	"fmt"
+	"github.com/nuts-foundation/nuts-node/network/dag/tree"
+	"go.etcd.io/bbolt"
+)
+
+const (
+	ibltBucketName = "treeIBLT"
+	xorBucketName  = "treeXOR"
+	// BucketFillPercent can be much higher than default 0.5 since the keys (unique node ids)
+	// should be added to the bucket in monotonic increasing order, and the values are of fixed size.
+	BucketFillPercent = 0.9
+)
+
+type bboltTree struct {
+	db *bbolt.DB
+}
+
+// NewBBoltTreeStore returns an instance of a BBolt based tree store. Buckets managed by this store are filled to BucketFillPercent
+func NewBBoltTreeStore(db *bbolt.DB) *bboltTree {
+	return &bboltTree{db: db}
+}
+
+// Read fills the tree with data in the bucket.
+// Returns an error the bucket does not exist, or if data in the bucket doesn't match the prototype.
+func (store bboltTree) Read(bucketName string, prototype tree.Data) (tree.Tree, error) {
+	tr := tree.New(prototype.New(), 0)
+	return tr, store.db.View(func(tx *bbolt.Tx) error {
+		// get bucket
+		bucket := tx.Bucket([]byte(bucketName))
+		if bucket == nil {
+			return fmt.Errorf("bucket '%s' not found", bucketName)
+		}
+
+		// get data
+		rawData := map[uint32][]byte{}
+		_ = bucket.ForEach(func(k, v []byte) error {
+			split := binary.LittleEndian.Uint32(k)
```

Comment:
we also have `byteOrder()` somewhere else

Reply:
The idea behind iblt.go#byteOrder() was to ensure Iblts use the same byte order everywhere so they can be easily compared across different platforms as is needed in set reconciliation.

I don't know if we want / need to extend this behavior further than the iblts